src/array.jl

```diff
@@ -694,14 +700,12 @@ function Base.push!(A::CategoricalVector, item)
     A
 end
 
-function Base.append!(A::CategoricalVector, B::CatArrOrSub)
-    levels!(A, union(levels(A), levels(B)))
+function Base.append!(A::CategoricalVector, B::AbstractArray)
     len = length(A)
     len2 = length(B)
     resize!(A.refs, len + len2)
-    for i = 1:len2
-        A[len + i] = B[i]
-    end
+    # As in Base, A will be left modified if it is not possible to copy B to A
```

Comment:
I wonder why Base doesn't call `resize!` in case of failure. That sounds trivial, undoing the operation is not as hard as in `copyto!`. But better stick to what Base does, and maybe file an issue in Julia.

Reply:
Maybe the reason is that you would have to do try-catch which will slow down the function? I will open an issue with Base.